tests/Domain/Classes/RouterTest.php

```diff
@@ -0,0 +1,255 @@
+
+        $serverRequestMock->expects($this->once())
+            ->method('getUri')->willReturn($uriInterfaceMock);
+
+        $uriInterfaceMock->expects($this->once())
+            ->method('getPath')->willReturn(self::ROUTE_PATH);
+
+        $serverRequestMock->expects($this->once())
+            ->method('getMethod')->willReturn('GET');
+
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage("Parameter '{$requiredParam}' is required");
+
+        $this->router->dispatch($serverRequestMock);
+    }
+
+    /**
+     * @throws NotFoundExceptionInterface
+     * @throws \ReflectionException
+     * @throws ContainerExceptionInterface
+     * @throws \JsonException
+     */
+    public function testDispatchInvalidMethod(): void
+    {
+        $_SERVER['HTTP_HOST']       = 'flexi';
+        $_SERVER['REQUEST_SCHEME']  = 'https';
+        $invalidMethod = 'POST';
+
+        $uriInterfaceMock = $this->createMock(UriInterface::class);
+        $serverRequestMock = $this->createMock(ServerRequestInterface::class);
+
+        $serverRequestMock->expects($this->once())
+            ->method('getUri')->willReturn($uriInterfaceMock);
+
+        $uriInterfaceMock->expects($this->once())
+            ->method('getPath')->willReturn(self::ROUTE_PATH);
+
+        $serverRequestMock->expects($this->exactly(2))
+            ->method('getMethod')->willReturn($invalidMethod);
+
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage("Method $invalidMethod is not allowed for this route");
+
+        $this->router->dispatch($serverRequestMock);
+    }
+
+    /**
+     * @throws NotFoundExceptionInterface
+     * @throws \ReflectionException
+     * @throws ContainerExceptionInterface
+     * @throws \JsonException
+     */
+    public function testDispatchDirectToNotFound(): void
+    {
+        $_SERVER['HTTP_HOST']       = 'flexi';
+        $_SERVER['REQUEST_SCHEME']  = 'https';
+
+        $uriInterfaceMock = $this->createMock(UriInterface::class);
+        $serverRequestMock = $this->createMock(ServerRequestInterface::class);
+        $responseInterfaceMock = $this->createMock(ResponseInterface::class);
+
+        $serverRequestMock->expects($this->once())
+            ->method('getUri')->willReturn($uriInterfaceMock);
+
+        $uriInterfaceMock->expects($this->once())
+            ->method('getPath')->willReturn('/invalid');
+
+        $this->event_bus->expects($this->exactly(2))
+            ->method('notify')->willReturnSelf();
+
+        $this->response_factory->expects($this->once())
+            ->method('createResponse')->willReturn($responseInterfaceMock);
+
+        $responseInterfaceMock->expects($this->once())
+            ->method('withHeader')->willReturnSelf();
+
+        $responseInterfaceMock->expects($this->once())
+            ->method('withStatus')->willReturnSelf();
+
+        $response = $this->router->dispatch($serverRequestMock);
```

Comment:
I have added an additional test called: `testDispatchDirectToNotFoundSpy `which I believe is the worse version because it assures `redirectToNotFound ` is called but does not go actually into the method. 

You can try it out yourself, following the steps:
1. Comment out: `testDispatchDirectToNotFound` 
2. Run with coverage
3. Check covered lines in `redirectToNotFound` 
4. Revert changes, run and check again.

I believe the current test is bettter, but we can have an extra.